cowprotocol/cowswap
# Enable the ProgressBar with the safe wallet

# Summary
Closes #794

Since the safe wallet API has a **delay** to report the **execution date**, it is proposed to update the `executionDate` when listening to the [ExecutionSuccess](https://github.com/safe-global/safe-contracts/blob/main/contracts/GnosisSafe.sol#L49) event of the Safe Contract.

⚠️ WIP ⚠️ 
- [x] Create a watcher to listen `ExecutionSuccess`
- [x] Create an updater that runs every time an order is created with a wallet safe
- [x] Overwrite `executionDate` in signed Order
- [x] Verify that the **ProgressBar** is started when the `executionDate` is updated
- [x] Avoid **ProgressBar** jumping when the state is delayed
- ~~Find out why in expert mode order falls as [RequestProviderError](https://github.com/cowprotocol/cowswap/blob/794-enable-progressbar/src/custom/utils/misc.ts#L145) when an order is placed with the same parameters~~

## To Test
- With a multi-signature wallet from the safe open to place an order
- Make the required signatures. 

The modal with the **progressBar** should start after listening to the execution event launched by the safe contract.
